src/main/java/org/prebid/server/bidder/consumable/ConsumableBidder.java

```diff
@@ -1,227 +1,261 @@
+                if (!isImpValid(bidRequest.getSite(), bidRequest.getApp(), impExt)) {
+                    continue;
+                }
+                if (Strings.isNullOrEmpty(placementId) && !Strings.isNullOrEmpty(impExt.getPlacementId())) {
+                    placementId = impExt.getPlacementId();
+                }
 
-        final Regs regs = request.getRegs();
+                imps.add(modifyImp(imp, impExt));
 
-        final String gpp = regs != null ? regs.getGpp() : null;
-        if (gpp != null) {
-            requestBuilder.gpp(gpp);
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
         }
-
-        final List<Integer> gppSid = regs != null ? regs.getGppSid() : null;
-        if (CollectionUtils.isNotEmpty(gppSid)) {
-            requestBuilder.gppSid(gppSid);
+        if (imps.isEmpty()) {
+            return Result.withErrors(errors);
         }
+        final BidRequest modRequest = modifyBidRequest(bidRequest, imps);
+        final String finalUrl = constructUri(placementId);
+        httpRequests.add(BidderUtil.defaultRequest(modRequest, resolveHeaders(), finalUrl, mapper));
+        return Result.of(httpRequests, errors);
+    }
 
-        final ExtRegs extRegs = regs != null ? regs.getExt() : null;
-        final String usPrivacy = extRegs != null ? extRegs.getUsPrivacy() : null;
-        if (usPrivacy != null) {
-            requestBuilder.usPrivacy(usPrivacy);
+    private ExtImpConsumable parseImpExt(Imp imp) {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), CONS_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage());
         }
+    }
 
-        final Integer gdpr = extRegs != null ? extRegs.getGdpr() : null;
-        final User user = request.getUser();
-        final ExtUser extUser = user != null ? user.getExt() : null;
-        final String gdprConsent = extUser != null ? extUser.getConsent() : null;
-        if (gdpr != null || gdprConsent != null) {
-            final ConsumableBidGdpr.ConsumableBidGdprBuilder bidGdprBuilder = ConsumableBidGdpr.builder();
-            if (gdpr != null) {
-                bidGdprBuilder.applies(gdpr != 0);
-            }
-            if (gdprConsent != null) {
-                bidGdprBuilder.consent(gdprConsent).build();
-            }
-            requestBuilder.gdpr(bidGdprBuilder.build());
+    private boolean isImpValid(Site site, App app, ExtImpConsumable impExt) {
+        return (app != null && !Strings.isNullOrEmpty(impExt.getPlacementId()))
+                || (site != null && impExt.getSiteId() != 0 && impExt.getNetworkId() != 0 && impExt.getUnitId() != 0);
+
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpConsumable impExt) {
+        final UpdateResult<ObjectNode> impExtUpdateResult = modifyImpExt(imp, impExt);
+
+        return imp;
+    }
```

Comment:
the update result is never used, shouldn't it be propagated to the imp? 

Also, I think using `UpdateResult` is not necessary, you can just return a new impExt and add it to the imp.

Also looks like this case is not covered by unit tests, so please cover it.

Reply:
This section has been removed as it didn't do anything anyways